app/models/concerns/account_settings.rb

```diff
@@ -110,8 +110,10 @@ def solr_collection_options
   end
   # rubocop:enable Metrics/BlockLength
 
-  def public_settings
-    all_settings.reject { |k, v| Account.private_settings.include?(k.to_s) || v[:disabled] }
+  def public_settings(is_superadmin: false)
```

Comment:
```suggestion
def public_settings(is_superadmin: false)
  settings = is_superadmin ? superadmin_settings(is_superadmin) : all_settings
  filter_out_private_and_disabled_settings(settings)
end

private

def filter_out_private_and_disabled_settings(settings)
  settings.reject do |key, value|
    Account.private_settings.include?(key.to_s) || value[:disabled]
  end
end

```

Reply:
^^ the above isn't quite right but something along those lines is what I was thinking